torchvision/transforms/functional.py

```diff
@@ -612,7 +612,38 @@ def adjust_saturation(img, saturation_factor):
     return img
 
 
-def adjust_hue(img, hue_factor):
+def _rgb2hsv(testing):
+    max_vals, _ = torch.max(testing, dim=0)
+    min_vals, _ = torch.min(testing, dim=0)
+    v = max_vals
+    s = (max_vals - min_vals) / max_vals
+    df = max_vals - min_vals
+    r, g, b = testing[0], testing[1], testing[2]
+    
+    
+    h = max_vals != min_vals
+    hr =  (max_vals == r) * ((g - b) / df) 
+    hg =  (max_vals == g) * ((b - r) / df + 2.0)
+    hb =  (max_vals == b) * ((r - g) / df + 4.0)    
+    h = h * (hr + hg + hb)
+    h = (h / 6.0) % 1.0
+    h[h != h] = 0
+    return torch.stack((h, s, v))
+ 
+def _hsv2rgb(testing):
+    h, s, v = testing[0], testing[1], testing[2]
+    i = (h * 6.0).type(torch.IntTensor)
+    f = (h * 6.0) - i
+    p = v * (1.0 - s)
+    q = v * (1.0 - f * s)
+    t = v * (1.0 - (1 - f) * s)
```

Comment:
I think some differences might be due to the `round` and `clip` behavior in https://github.com/python-pillow/Pillow/blob/14859ce5068ecfa750097122b61823b44d618892/src/libImaging/Convert.c#L401-L406
What about performing the operations in float, but using `torch.round` and `torch.clamp(tensor, 0, 255)`?